doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+a painful breaking change.
+
+### Why SQL?
+
+The idea of the region API allowing only SQL queries has been met with some
+skepticism. Here are the arguments:
+
+* There is only one stable API. We don't need to worry about a separate HTTP API
+  for the web UI that needs its own versioning and backwards compatibility
+  guarantees. We just keep the SQL language stable and the system catalog
+  stable, which we already have to do to avoid breaking our customers. The web
+  UI then gets to piggyback on this stability for free.
+
+* Unified access control. The same SQL access control rules that regulate what
+  actions a user can take from a SQL shell regulate what actions they can take
+  from the web UI.
+
+* We get extremely valuable experience using Materialize to power a web UI,
+  which is exactly the use case we're selling.
+
+  The APIs we develop to support our own web UI, like tailing cluster
+  provisioning updates over WebSockets, are APIs that will also be useful to our
+  customers.
+
+I'd like to better understand the arguments against this approach. Here is what
+I recall:
+
+  * SQL won't integrate well with monitoring tools.
+
+    Counterpoint: we're going to need to get good at monitoring SQL queries in
+    order to support our customers.
+
+  * SQL won't integrate well with TypeScript.
+
+    Counterpoint: Nikhil will personally guarantee that we can generate
+    TypeScript bindings for the SQL API.
+
+#### SQL access control
+
+The approach of integrating Materialize's custom object types into the SQL
+access control system is modeled after [Snowflake's Access Control
+Privileges][snowflake-acl].
+
+I like this approach for several reasons:
+
+  * We don't need to invent a new access control model.
+
+  * We'll likely need to build support for `GRANT` and `REVOKE` for
+    PostgreSQL compatibility no matter what. If we invented a new access control
+    model, we'd have to figure out how to integrate it with `GRANT` and
+    `REVOKE`.
+
+  * Snowflake has also extended the SQL standard with Snowflake-specific
+    objects. The fact that they've managed to integrate those objects with the
+    SQL permission model is proof that this can be made to work.
+
+My biggest gripe with SQL access control is object ownership. The concept of a
+single user owning an object has never seemed particularly useful to me, and
+can be actively frustrating when trying to drop or rename an object. We need
+to evaluate whether we want to support this.
+
+TODO: engage @andrioni.
+
+### Frontegg
+
+There are a few rough edges around the user experience of authentication that
+are a result of using Frontegg as our authentication provider. In particular,
+[app-specific passwords](#app-specific-password) and the separation between
+account-level roles and region-level roles are quite confusing, and the
+inability to change a user's email address is an annoying limitation.
+
+I don't see a better alternative than what's proposed if we want to keep
```

Comment:
> Using Frontegg accounts may force customers to tightly-couple database access to the structure of their SSO accounts.

Well, this is the whole reason to use SSO! You use SSO because you want your SSO provider to be responsible for automatically provisioning accounts in Materialize Platform. If we require a manual step to actually activate that user account, we've erased the value of SSO. 

> I also don't like the ergonomics around using API tokens as SQL passwords. It isn't great and will cause friction. 

I don't either, but I think the choice is between a) the somewhat strange integration I've proposed here and b) abandoning Frontegg. Avoiding a distinction between "cloud accounts" and "database accounts" is one of the key UX features for Materialize Platform. Having two separate notions of accounts is a dead giveaway that we're just taking an on-prem product and running it in the cloud.

If we were to abandon Frontegg, we could do something similar to what Snowflake did with [`CREATE SECURITY INTEGRATION`](https://docs.snowflake.com/en/sql-reference/sql/create-security-integration.html).

> I suspect one of the first feature requests we're going to get is the ability to specify a password.

What makes you say this? I've seen plenty of cloud products that only give you the option to regenerate the password, not specify one from scratch.